### ISSUE TITLE
Odoo CSP (Content-Security-Policy) without unsafe-line and unsafe-eval

### ISSUE THREAD

**Issue by danger89** [odoo/odoo#22404]
Impacted versions: All

Steps to reproduce: Enable CSP (Content-Security-Policy) on server side like Nginx/Apache server.

Current behavior: Web pages will not be loaded when `script-src 'self' 'unsafe-inline' 'unsafe-eval'` is omitted from the Content-Security-Policy header.

Expected behavior: Odoo should work even with CSP enabled without `unsafe-inline` and `unsafe-eval`.

[Read more...](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src)

Video/Screenshot link (optional):
![afbeelding](https://user-images.githubusercontent.com/628926/35190299-3467429a-fe5e-11e7-9d67-f46bac0d1954.png)

**Comment by greaber** [comment]
What is the strictest CSP that can be used with Odoo currently?

**Comment by TimLai125** [comment]
@danger89 Yes, there are quite amount of incline scripts within Odoo that make the CSP quite difficult to setup.
While it seems that the CSP header is also not being set in https://www.odoo.com
![image](https://user-images.githubusercontent.com/27185427/42557998-8b2bfecc-8522-11e8-84fa-4bde61f0fac2.png)

@greaber, Hi I am new to this area but I am afraid that you need to set `unsafe-inline` to the CSP (which no really secured in sense) for now in the Odoo website. Please let me know if you have any good suggestion on this.

**Comment by greaber** [comment]
I'm not very experienced with CSP, but if the inline scripts are all static (not created from JavaScript strings), it should be possible to at least use unsafe-eval.

**Comment by mtancoigne** [comment]
Any news on this one ? Is it possible from Odoo to make a statement about the _best_ configuration concerning the CSPs ?

**Comment by iambkramer** [comment]
Checking to see if there is any update from Odoo?

**Comment by TadeuszKarpinski** [comment]
I am waiting for this change

**Comment by mcassuto** [comment]
+1 one our customer has been audited by a security firm.... we need this as well

**Comment by matteoopenf** [comment]
Hi, there is any news?

**Comment by rauldb-br** [comment]
Hello,
¿anyone know something?

**Comment by matteoopenf** [comment]
> Hello,
> 
> ¿anyone know something?

We also wait update

**Comment by rauldb-br** [comment]
For use CSP without `'unsafe-inline' 'unsafe-eval'` odoo have to configurate `script-src 'nonce-rAnd0m'` or `cript-src 'unsafe-hashes' 'sha256-***'`.
https://content-security-policy.com/nonce/
https://content-security-policy.com/unsafe-hashes/
https://content-security-policy.com/strict-dynamic/

¿This issue is only for CE? My Odoo.sh not have that problem.

**Comment by spadae22** [comment]
We have the same issue, odoo 14 enterprise on AWS

**Comment by spadae22** [comment]
I tried this: Well I got this: add_header Content-Security-Policy "default-src 'self'; script-src 'self' 'unsafe-inline'; style-src 'self' 'unsafe-inline' fonts.googleapis.com fonts.gstatic.com https://fonts.odoocdn.com/; font-src 'self' data: fonts.googleapis.com fonts.gstatic.com https://fonts.odoocdn.com/; img-src 'self' data:; connect-src 'self';";

Then the screen is blank and I get:
<img width="1006" alt="Screenshot 2023-11-07 at 12 19 41 PM" src="https://github.com/odoo/odoo/assets/7741100/0e09bca2-7b37-4453-8da5-748da74eecc4">

**Comment by df-obs** [comment]
Any news on this? It would be great to have a best-practice from Odoo for the most secure configuration that still works with Odoo

**Comment by ahmed-aly-aut** [comment]
Any news?

**Comment by pangeranpas** [comment]
Any news?
